src/sidebar/util/annotation-metadata.js

```diff
@@ -306,3 +306,36 @@ export function quote(annotation) {
   const quoteSel = target.selector.find(s => s.type === 'TextQuoteSelector');
   return quoteSel ? /** @type {TextQuoteSelector}*/ (quoteSel).exact : null;
 }
+
+/**
+ * Has this annotation been edited subsequent to its creation?
+ *
+ * @param {Annotation} annotation
+ * @return {boolean}
+ */
+export function hasBeenEdited(annotation) {
+  let hasBeenEdited = false;
+  if (!annotation.updated || !annotation.created) {
+    return hasBeenEdited;
+  }
+  // updated and created are strings
+  const maybeEdited =
+    annotation.updated && annotation.created !== annotation.updated;
+  if (maybeEdited) {
+    // Both updated and created SHOULD be ISO-8601-formatted
+    // with microsecond resolution; (NB: Date.prototype.getTime() returns
+    // milliseconds since epoch, so we're dealing in ms after this)
+    const created = new Date(annotation.created).getTime();
+    const updated = new Date(annotation.updated).getTime();
+    if (isNaN(created) || isNaN(updated)) {
+      // If either is not a valid date...
+      return hasBeenEdited;
+    }
+    if (updated - created > 2000) {
+      // A 2-second grace period here to avoid any service-side small
+      // disparities between created and updated
+      hasBeenEdited = true;
+    }
+  }
+  return hasBeenEdited;
```

Comment:
This function can be streamlined / simplified in a few ways

- You can avoid tracking the `hasBeenEdited` state through the function given that all the early returns always return a fixed value
- If `annotation.updated == annotation.created` you can return early
- The `annotation.updated && annotation.created !== annotation.updated` state can be simplified to `annotation.created !== annotation.updated` given that the test above implies that `annotation.updated` will not be falsey.

```js
export function hasBeenEdited(annotation) {
  if (!annotation.updated || !annotation.created) {
    // Annotation has not yet been saved.
    return false;
  }

  if (annotation.created === annotation.updated) {
    return false;
  }

  // Both updated and created SHOULD be ISO-8601-formatted
  // with microsecond resolution; (NB: Date.prototype.getTime() returns
  // milliseconds since epoch, so we're dealing in ms after this)
  const created = new Date(annotation.created).getTime();
  const updated = new Date(annotation.updated).getTime();
  if (isNaN(created) || isNaN(updated)) {
    // If either is not a valid date...
    return false;
  }

  // A 2-second grace period here to avoid any service-side small
  // disparities between created and updated
  return updated - created > 2000;
}
```